components/eventing-controller/controllers/subscription/nats/integration_test.go

```diff
@@ -0,0 +1,837 @@
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveCleanEventTypes([]string{
+						cleanEventType("0alpha"),
+						cleanEventType("1alpha"),
+					}),
+				},
+			},
+		},
+
+		{
+			name: "clean event types; delete a filter",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, uncleanEventType("0")),
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, uncleanEventType("1")),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURLFromSvc(ens.subscriberSvc),
+			},
+			expectBefore: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveCleanEventTypes([]string{
+						cleanEventType("0"),
+						cleanEventType("1"),
+					}),
+				},
+			},
+			changeSubscription: func(subscription *eventingv1alpha1.Subscription) {
+				subscription.Spec.Filter.Filters = subscription.Spec.Filter.Filters[:1]
+			},
+			expectAfter: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveCleanEventTypes([]string{
+						cleanEventType("0"),
+					}),
+				},
+			},
+		},
+
+		{
+			name: "change configuration",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, uncleanEventType("")),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURLFromSvc(ens.subscriberSvc),
+			},
+			expectBefore: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+				},
+			},
+			changeSubscription: func(subscription *eventingv1alpha1.Subscription) {
+				subscription.Spec.Config = &eventingv1alpha1.SubscriptionConfig{
+					MaxInFlightMessages: 101,
+				}
+			},
+			expectAfter: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(
+						&eventingv1alpha1.SubscriptionConfig{
+							MaxInFlightMessages: 101,
+						},
+					),
+				},
+				natsSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.BeNotNil(),
+					reconcilertesting.HaveSubject(cleanEventType("")),
+					reconcilertesting.BeValid(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "resolve multiple conditions",
```

Comment:
I wonder why this is not tested inside `TestCreateSubscription` instead. I don't see the need to change something in order to test that multiple conditions are cleaned up and that there is only one condition at the end.

Reply:
The cleaning of `OrderCreatedEventTypeNotClean` can be tested without testing the multiple conditions IMHO.

Reply:
this test case is special. it was the last one i moved over from the old tests, because I also struggled with understanding, what the goal of this test actually was.

Reply:
we can sync on this, if you want

Reply:
this is for a followup pr